### PR TITLE
DEP: Enforce set_values and set_codes inplace and positional args deprecation

### PR DESCRIPTION
**Pull request by phofl**
#41485 #35626

### CODE REVIEW

**Review by MarcoGorelli — APPROVED** [review]
looks good to me

File: pandas/tests/indexes/multi/test_compat.py [hunk 1]
```diff
@@ -91,8 +89,7 @@ def test_inplace_mutation_resets_values():
     tm.assert_almost_equal(exp_values, new_values)
 
     # ...and again setting inplace should drop _values from _cache, etc
```
Comment (WillAyd) [inline pandas/tests/indexes/multi/test_compat.py:91]: Minor but can delete this line

File: pandas/tests/indexes/multi/test_compat.py [hunk 2]
```diff
@@ -65,9 +65,7 @@ def test_inplace_mutation_resets_values():
     # ...and values is still same too
     tm.assert_almost_equal(mi1.values, vals)
 
-    # Inplace should drop _values from _cache
-    with tm.assert_produces_warning(FutureWarning):
-        mi1.set_levels(levels2, inplace=True)
+    mi1 = mi1.set_levels(levels2)
```
Comment (WillAyd) [inline pandas/tests/indexes/multi/test_compat.py:68]: Do we even need this part of the test any more? On initial glance seems duplicative of what's already in the test and if so would rather delete

File: pandas/tests/indexes/multi/test_equivalence.py [hunk 3]
```diff
@@ -254,12 +251,10 @@ def test_is_():
     mi4 = mi3.view()
 
     # GH 17464 - Remove duplicate MultiIndex levels
-    with tm.assert_produces_warning(FutureWarning):
-        mi4.set_levels([list(range(10)), list(range(10))], inplace=True)
+    mi4 = mi4.set_levels([list(range(10)), list(range(10))])
```
Comment (WillAyd) [inline pandas/tests/indexes/multi/test_equivalence.py:254]: Similar comment - might be able to just remove

**Review by WillAyd — CHANGES_REQUESTED** [review]
minor items otherwise lgtm

File: pandas/tests/indexes/multi/test_compat.py (hunk 1 above)
Comment (phofl) [reply pandas/tests/indexes/multi/test_compat.py:91]: Done

File: pandas/tests/indexes/multi/test_compat.py (hunk 2 above)
Comment (phofl) [reply pandas/tests/indexes/multi/test_compat.py:68]: Good point, removed

File: pandas/tests/indexes/multi/test_equivalence.py (hunk 3 above)
Comment (phofl) [reply pandas/tests/indexes/multi/test_equivalence.py:254]: This is a bit different, would like to keep it

File: doc/source/whatsnew/v2.0.0.rst [hunk 4]
```diff
@@ -145,7 +145,8 @@ Deprecations
 Removal of prior version deprecations/changes
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 - Disallow passing non-round floats to :class:`Timestamp` with ``unit="M"`` or ``unit="Y"`` (:issue:`47266`)
--
+- Remove argument ``inplace`` from :meth:`MultiIndex.set_levels` and :meth:`MultiIndex.set_codes` (:issue:`35626`)
+- Disallow passing positional arguments to :meth:`MultiIndex.set_levels` and :meth:`MultiIndex.set_codes` (:issue:`41485`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.0.0.rst:149]: Could you add a test where `set_levels` and `set_codes` raises when passed positional args?

**Review by WillAyd — APPROVED** [review]
lgtm @mroeschke merge when happy

File: doc/source/whatsnew/v2.0.0.rst (hunk 4 above)
Comment (phofl) [reply doc/source/whatsnew/v2.0.0.rst:149]: Added

**Comment by mroeschke** [comment]
Thanks @phofl

**Pull request merged**